data/org.freedesktop.portal.GlobalShortcuts.xml

```diff
@@ -0,0 +1,188 @@
+              object path element. See the #org.freedesktop.portal.Session documentation for
+              more information about the session handle.
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+
+        The following results get returned via the #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>session_handle o</term>
+            <listitem><para>
+              The session handle. An object path for the
+              #org.freedesktop.portal.Session object representing the created
+              session.
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+    -->
+    <method name="CreateSession">
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="o" name="handle" direction="out"/>
+      <annotation name="org.qtproject.QtDBus.QtTypeName.In0" value="QVariantMap"/>
+    </method>
+
+      <!--
+        ListShortcuts:
+        @session_handle: Object path for the #org.freedesktop.impl.portal.Session object representing the session
+        @options: Unused so far
+
+        List all the shortcuts with their invocation descriptions.
+
+        The following results get returned via the #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>shortcuts a(sa{sv})</term>
+            <listitem><para>
+              The different shortcut ids that have been registered. They are mapped to a list of pairs including:
+                - description: User-readable text describing what the shortcut does
+                - trigger: User-readable text describing how to trigger the shortcut
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+    -->
+    <method name="ListShortcuts">
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="o" name="request_handle" direction="out"/>
+    </method>
+
+    <!--
+        BindShortcuts:
+        @session_handle: Object path for the #org.freedesktop.impl.portal.Session object representing the session
+        @shortcuts: The identifier of the shortcuts we intend to register, empty for all shortcuts
+        @request_handle: Object path for the #org.freedesktop.portal.Request object representing this call
+
+        Shows a dialog with all shortcuts to configure and offers them back to the app when done.
+
+        The following results get returned via the #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>shortcuts a(sa{sv})</term>
+            <listitem><para>
+              The different shortcut ids that have been registered. They are mapped to a list of pairs including:
+                - description: User-readable text describing what the shortcut does
+                - trigger: User-readable text describing how to trigger the shortcut
```

Comment:
What's the purpose of repeating the same information that was provided when creating the session?

If the assumption is that one creates a session consisting of a predefined set of shortcuts, and a application wanting to request that a subset of them to be bound, it should be enough to just provide a list of shortcut IDs; the rest is already there.

Reply:
The idea was that we'd get the triggers that were just bound. I agree that getting the descriptions again is not that useful, I just added it so that we get the shortcuts dictionary every time rather than several very similar ones.

After implementing, it seems to me that it might be too synchronous an API.

Reply:
Maybe it should be documented in `ListShortcuts` and just referred to from here?

Reply:
Referring to ListShortcuts now.